op-challenger/game/fault/trace/vm/executor.go

```diff
@@ -33,8 +33,13 @@ type Config struct {
 	InfoFreq         uint // Frequency of progress log messages (in VM instructions)
 }
 
+type VmConfig interface {
+	Cfg() Config
```

Comment:
I really don't get why we're exposing the global config from this.  It expands the scope of `VmConfig` to be bigger than it really should be.  I think maybe we're optimising to minimise the number of params to `NewExecutor` but that feels wrong to me and if anything we should be reducing dependencies on the full `Config` and moving to more situation specific, focussed confirms (like `VmConfig` would be without this method).